### PR TITLE
Add healthchecks

### PR DESCRIPTION
**Pull request by omegas27**

### CODE REVIEW

**Comment by omegas27** [comment]
Tested on gorgoth

File: helm/promxy-app/templates/deployment.yaml [hunk 1]
```diff
@@ -27,13 +27,42 @@ spec:
       containers:
       - name: {{ include "name" . }}
         image: "{{ .Values.Installation.V1.Registry.Domain }}/{{ .Values.promxy.image.name }}:{{ .Values.promxy.image.tag }}"
-        ports:
-        - containerPort: {{ .Values.promxy.port }}
         args:
+        - "--bind-addr=:{{ .Values.promxy.port }}"
         - "--config={{ .Values.promxy.mountPath }}/config.yaml"
         - "--web.enable-lifecycle"
+        # - "--web.external-url"
```
Comment (mig4) [inline helm/promxy-app/templates/deployment.yaml:34]: Was this meant to have a value here?

File: helm/promxy-app/templates/configmap.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ include "resource.default.name" . }}
+  name: {{ include "resource.configmap.name" . }}
```
Comment (mig4) [inline helm/promxy-app/templates/configmap.yaml:4]: Just out of curiosity why rename it?
Cause then to reference this resource you'd have `configmap/promxy-configmap` which is repetitive.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add healthchecks and allow port configuration
```
Comment (mig4) [inline CHANGELOG.md:12]:
```suggestion
- Added healthchecks and allow port configuration
- Define resource requests/limits to help with scheduling
- Set monitoring labels and annotations so the service is picked up by Giant Swarm monitoring
```

**Review by mig4 — CHANGES_REQUESTED** [review]
Looks good, just one suggestion and a couple of questions.

File: helm/promxy-app/templates/deployment.yaml (hunk 1 above)
Comment (omegas27) [reply helm/promxy-app/templates/deployment.yaml:34]: This is to not forget as we will need to set that when we create the ingress for it

File: helm/promxy-app/templates/configmap.yaml (hunk 2 above)
Comment (omegas27) [reply helm/promxy-app/templates/configmap.yaml:4]: I can change this sure :) I renamed that as we have this for networkpolicies, psp and so on

Comment (omegas27) [reply helm/promxy-app/templates/configmap.yaml]: I changed that to resource.default.name

**Review by mig4 — APPROVED** [review]
❤️

**Pull request merged**